File: icamera/src/main/java/me/shouheng/icamera/enums/MediaType.java
package me.shouheng.icamera.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static me.shouheng.icamera.enums.MediaType.*;

/**
 * Media type
 * 
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2019-12-28 16:29
 */
@IntDef(value = {TYPE_PICTURE, TYPE_VIDEO})
@Retention(value = RetentionPolicy.SOURCE)
public @interface MediaType {

    /** Picture */
    int TYPE_PICTURE    = 0;

    /** Video */
    int TYPE_VIDEO      = 1;
}

File: icamera/src/main/java/me/shouheng/icamera/enums/MediaType.kt
package me.shouheng.icamera.enums

import androidx.annotation.IntDef

/**
 * Media type
 *
 * @author [WngShhng](mailto:shouheng2015@gmail.com)
 * @version 2019-12-28 16:29
 */
@IntDef(value = [MediaType.TYPE_PICTURE, MediaType.TYPE_VIDEO])
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class MediaType {
    companion object {
        /** Picture  */
        const val TYPE_PICTURE = 0
        /** Video  */
        const val TYPE_VIDEO = 1
    }
}